app/javascript/app/components/emission-pathways-model-table/emission-pathways-model-table-selectors.js

```diff
@@ -0,0 +1,34 @@
+import { createSelector } from 'reselect';
+import isEmpty from 'lodash/isEmpty';
+import pick from 'lodash/pick';
+
+const getCategoryName = data => data.category;
+const getId = data => data.id || null;
+const getData = espData =>
+  (espData.espModelsData && !isEmpty(espData.espModelsData.data)
+    ? espData.espModelsData.data
+    : null);
+const getIdData = createSelector([getData, getId], (data, id) => {
+  if (!data || !id) return null;
+  const idData = data.find(d => String(d.id) === id);
+  return idData || null;
+});
+
+export const filteredCategoryData = createSelector(
+  [getIdData, getCategoryName],
+  (data, category) => {
+    if (!data) return null;
+    const categoryName = category.toLowerCase();
+    const categoryWhiteListedFields = {
+      Scenarios: ['name', 'category', 'description'],
```

Comment:
Can you check with JC that extra fields aren't wanted. Just incase the column selector was actually wanted.

Reply:
Actually just seen in the designs that the column selector is required. Can you get these as defaults and then allow all columns to be chosen.